.github/workflows/ci-workflow.yml

```diff
@@ -0,0 +1,158 @@
+name: Continuous Integration
+on: [push, pull_request]
+
+env:
+  # Share node version across jobs
+  # actions/setup-node doesn't currently look at any config files
+  # for now we do the simplest thing and set an environment variable.
+  # See https://github.com/actions/setup-node/pull/239
+  # Maybe we could use https://github.com/guardian/actions-setup-node directly in the meantime?
+  NODE_VERSION: '14'
+
+jobs:
+  node-test:
+    runs-on: ubuntu-20.04
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v2
+        with:
+          node-version: ${{ env.NODE_VERSION }}
+      # Cache and install npm dependencies using https://github.com/bahmutov/npm-install
+      - uses: bahmutov/npm-install@v1
+      - run: npm test
+
+  ruby-lint:
+    runs-on: ubuntu-20.04
+    steps:
+      - uses: actions/checkout@v2
+      - uses: ruby/setup-ruby@v1
+        with:
+          bundler-cache: true
+      - run: bundle exec rake ruby:lint
+
+  # Actions can run in parallel at the job or the workflow level.
+  # In order to run our build-dependent tests in parallel we need a way
+  # to build the docs and then "seed" each stage with a copy.
+  # We can do this using using artifacts along with a `needs` clause on each job
+  # See: https://docs.github.com/en/actions/learn-github-actions/essential-features-of-github-actions#sharing-data-between-jobs
+  #      https://docs.github.com/en/actions/learn-github-actions/managing-complex-workflows#creating-dependent-jobs
+  build:
+    runs-on: ubuntu-20.04
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v2
+        with:
+          node-version: ${{ env.NODE_VERSION }}
+      # Cache npm dependencies using https://github.com/bahmutov/npm-install
+      - uses: bahmutov/npm-install@v1
+      - uses: ruby/setup-ruby@v1
+        with:
+          bundler-cache: true
+      - run: npm run build --if-present
+      - run: npm run docs:build
+      - uses: actions/upload-artifact@v2
+        with:
+          name: docs-build
+          path: |
+            docs
+            lib
+          retention-days: 1
+
+  pa11y:
+    needs: build
+    runs-on: ubuntu-20.04
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v2
+        with:
+          node-version: ${{ env.NODE_VERSION }}
+      # Cache and install npm dependencies using https://github.com/bahmutov/npm-install
+      - uses: bahmutov/npm-install@v1
+      - uses: actions/download-artifact@v2
+        with:
+          name: docs-build
+      # Explicitly install testing dependencies
+      -  run: npm ci
+        working-directory: ./testing
+      - run: ruby -run -e httpd -- docs -p 6006 &
```

Comment:
This is not the most elegant, but realised we can use the built in ruby to start a simple server in the background. This can be consistent for each steps as it doesn't rely on having all our dependencies installed for every job.